src/CarbonAware.DataSources/CarbonAware.DataSources.ElectricityMaps/src/Client/ElectricityMapsClient.cs

```diff
@@ -103,6 +94,72 @@ public async Task<ForecastedCarbonIntensityData> GetForecastedCarbonIntensityAsy
         return await GetCurrentForecastAsync(parameters);
     }
 
+    public async Task<PastRange> GetPastRangeDataAsync(string latitude, string longitude, DateTimeOffset startTime, DateTimeOffset endTime)
+    {
+        _log.LogDebug("Requesting carbon intensity using latitude {latitude} longitude {longitude}",
+           latitude, longitude);
+
+        var parameters = new Dictionary<string, string>()
+        {
+            { QueryStrings.Latitude, latitude },
+            { QueryStrings.Longitude, longitude },
+            { QueryStrings.StartTime, DateTimeToString(startTime) },
+            { QueryStrings.EndTime, DateTimeToString(endTime) },
+        };
+
+        return await GetPastRangeDataAsync(parameters);
+    }
+
+    public async Task<PastRange> GetPastRangeDataAsync(string zone, DateTimeOffset startTime, DateTimeOffset endTime)
+    {
+        _log.LogDebug("Requesting carbon intensity using zone {zone}",
+           zone);
+
+        var parameters = new Dictionary<string, string>()
+        {
+            { QueryStrings.ZoneName, zone },
+            { QueryStrings.StartTime, DateTimeToString(startTime) },
+            { QueryStrings.EndTime, DateTimeToString(endTime) },
+        };
+
+        return await GetPastRangeDataAsync(parameters);
+    }
+    
+    private static string DateTimeToString(DateTimeOffset dateTime)
+    {
+        return dateTime.ToUniversalTime().ToString("O", CultureInfo.InvariantCulture);
+    }
+
+    // Internal call to check for allowed zones and then make GET request to History endpoint
+    private async Task<HistoryCarbonIntensityData> GetHistoryCarbonIntensityDataAsync(Dictionary<string, string> parameters)
+    {
+        await CheckZonesAllowedForPath(Paths.History, parameters);
+        AddOptionalParameters(parameters);
+
+        using (var result = await this.MakeRequestGetStreamAsync(Paths.History, parameters))
+        {
+            return await JsonSerializer.DeserializeAsync<HistoryCarbonIntensityData>(result, _options) ?? throw new ElectricityMapsClientException($"Error getting history carbon intensity data");
+        }
+    }
+    private void AddOptionalParameters(Dictionary<string, string> parameters)
+    {
+        if (_configuration.EmissionFactorType != null)
```

Comment:
Thats a good point. However, after some analysis we decided to keep all these as strings to be consistent. This is because -
If we have it as enum, and a value other than 'lifecycle' or 'direct' is provided, the auto binding fails causing it to throw a 500 even before a call is made to EM. If we keep it as string, and provide an incorrect value, EM sends back an error stating that the value of emissionFactor was incorrect and we show the same to the end user. This error is much clearer than the Internal server error.